lib/surface/components/link.ex

```diff
@@ -27,37 +28,68 @@ defmodule Surface.Components.Link do
 
   use Surface.Component
 
-  import Phoenix.HTML.Link, only: [link: 2]
-
-  @doc "Place to link to"
+  @doc "The page to link to"
   prop to, :string, required: true
 
+  @doc "The method to use with the link"
+  prop method, :atom, default: :get
+
   @doc "Class or classes to apply to the link"
   prop class, :css_class
 
-  @doc "Keyword with options to be passed down to `link/2`"
-  prop opts, :keyword, default: []
-
   @doc """
-  The label for the generated `<a>` alement, if no content (default slot) is
-  provided.
+  The label for the generated `<a>` alement, if no content (default slot) is provided.
   """
   prop label, :string
 
   @doc "Triggered on click"
   prop click, :event
 
+  @doc """
+  Additional attributes to add onto the generated element
+  """
+  prop opts, :keyword, default: []
+
   @doc """
   The content of the generated `<a>` element. If no content is provided,
   the value of property `label` is used instead.
   """
   slot default
 
   def render(assigns) do
-    children = ~H"<slot>{{ @label }}</slot>"
-
     ~H"""
-    {{ link [to: @to] ++ prop_to_attr_opts(@class, :class) ++ @opts ++ event_to_opts(@click, :phx_click), do: children }}
+    <a
+      class={{ @class }}
+      href={{ @to }}
+      :attrs={{ @opts ++ event_to_opts(@click, :phx_click) |> opts_to_attrs(assigns) }}
+    ><slot>{{ @label }}</slot></a>
     """
   end
+
+  defp opts_to_attrs(opts, assigns) do
+    for {key, value} <- opts do
+      case key do
+        :csrf_token -> {:"data-csrf", value}
+        :phx_click -> {:"phx-click", value}
+        :phx_target -> {:"phx-target", value}
+        :method -> method_to_attrs(value, assigns.to)
+        :data -> data_to_attrs(value)
+        _ -> {key, value}
+      end
+    end
+    |> List.flatten()
+  end
```

Comment:
I think this is ok for now. I'm working on a PR for `phoenix_html` that will expose a function to generate attributes consistently. As soon as it gets merged, we'll replace the `:attrs` implementation to use it, which will also fix #255.